catalog/imaging/imaging-plugin-nitf/src/main/java/org/codice/alliance/plugin/nitf/NitfPostIngestPlugin.java

```diff
@@ -243,37 +287,29 @@ private BufferedImage renderImageUsingOriginalDataModel(ContentItem contentItem)
   }
 
   private BufferedImage render(
-      ContentItem contentItem,
+      InputStream inputStream,
       Function<Pair<ImageSegment, NitfRenderer>, BufferedImage> imageSegmentFunction)
-      throws IOException, ParseException, NitfFormatException {
+      throws NitfFormatException {
 
     final ThreadLocal<BufferedImage> bufferedImage = new ThreadLocal<>();
 
-    if (contentItem != null) {
-      InputStream inputStream = contentItem.getInputStream();
-
-      if (inputStream != null) {
-        try {
-          NitfRenderer renderer = getNitfRenderer();
-
-          new NitfParserInputFlowImpl()
-              .inputStream(inputStream)
-              .allData()
-              .forEachImageSegment(
-                  segment -> {
-                    if (bufferedImage.get() == null) {
-                      BufferedImage bi =
-                          imageSegmentFunction.apply(new ImmutablePair<>(segment, renderer));
-                      if (bi != null) {
-                        bufferedImage.set(bi);
-                      }
-                    }
-                  })
-              .end();
-        } finally {
-          IOUtils.closeQuietly(inputStream);
-        }
-      }
+    if (inputStream != null) {
+      NitfRenderer renderer = getNitfRenderer();
+
+      new NitfParserInputFlowImpl()
+          .inputStream(inputStream)
+          .allData()
```

Comment:
:exclamation: This was written before the `org.codice.alliance.imaging.nitf.api.NitfParserService` was created. Now that we have that, rather than calling `new NitfParserInputFlowImpl()` I think we'd be better off wiring in `NitfParserService` from Blueprint and calling that.

Reply:
Switched to using the NitfParserService